airbyte-integrations/connectors/source-freshdesk/source_freshdesk/streams.py

```diff
@@ -0,0 +1,254 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional
+from urllib import parse
+
+import pendulum
+import requests
+import re
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.streams.core import IncrementalMixin
+from airbyte_cdk.sources.streams.http import HttpStream
+from requests.auth import AuthBase
+from source_freshdesk.utils import CallCredit
+
+
+class FreshdeskStream(HttpStream, ABC):
+    """Basic stream API that allows to iterate over entities"""
+    call_credit = 1  # see https://developers.freshdesk.com/api/#embedding
+    result_return_limit = 100
+    primary_key = "id"
+    link_regex = re.compile(r'<(.*?)>;\s*rel="next"')
+
+    def __init__(self, authenticator: AuthBase, config: Mapping[str, Any], *args, **kwargs):
+        super().__init__(authenticator=authenticator)
+        requests_per_minute = config["requests_per_minute"]
+        start_date = config["start_date"]
+        self.domain = config["domain"]
+        self._call_credit = CallCredit(balance=requests_per_minute) if requests_per_minute else None
+        # By default, only tickets that have been created within the past 30 days will be returned.
+        # Since this logic rely not on updated tickets, it can break tickets dependant streams - conversations.
+        # So updated_since parameter will be always used in tickets streams. And start_date will be used too
+        # with default value 30 days look back.
+        self.start_date = pendulum.parse(start_date) if start_date else pendulum.now() - pendulum.duration(days=30)
+    
+    @property
+    def url_base(self) -> str:
+        return parse.urljoin(f"https://{self.domain.rstrip('/')}", "/api/v2")
+    
+    def backoff_time(self, response: requests.Response) -> Optional[float]:
+        if response.status_code == requests.codes.too_many_requests:
+            return float(response.headers.get("Retry-After"))
```

Comment:
```suggestion
            return float(response.headers.get("Retry-After", 0))
```
You will have a `TypeError` if `Retry-After` is not in the headers: `float(None) -> TypeError`

Reply:
Good catch! Done.